include/azure_kinect_ros_driver/k4a_exposure_calibration.h

```diff
@@ -0,0 +1,158 @@
+// PlusOne Robotics
+// Author: Shannon Stoehr
+// email:  shannon.stoehr@plusonerobotics.com
+
+#pragma once
+
+// Library headers
+#include <ros/ros.h>
+#include <sensor_msgs/Image.h>
+#include <image_transport/image_transport.h>
+#include <cv_bridge/cv_bridge.h>
+#include <opencv2/opencv.hpp>
+#include <dynamic_reconfigure/client.h>
+#include <k4a/k4a.h>
+#include <mutex>
+
+// Project headers
+#include "azure_kinect_ros_driver/AzureKinectParamsConfig.h"
+#include "azure_kinect_ros_driver/k4a_update_exposure.h"
+#include "azure_kinect_ros_driver/k4a_auto_tune_exposure.h"
+#include "azure_kinect_ros_driver/k4aCameraExposureServiceErrorCode.h"
+
+/**
+ * @brief Class that allows the user to tune exposure in real time from the command line
+ */
+class K4AExposureCalibration
+{
+public:
+    /** @brief default constructor and destructor*/
+    K4AExposureCalibration();
+    ~K4AExposureCalibration();
+
+    /**
+     * @brief initializes calibrator with a provided node handle
+     * @details Subscribes the node to /rgb/raw/image.
+     *          Advertises the k4a_update_exposure service and the k4a_auto_tune_exposure service.
+     * @param[in] nh ROS node handle
+     */
+    K4AExposureCalibration(ros::NodeHandle& nh);
+
+    /**
+     * @brief call k4a_nodelet_manager/set_parameters to update exposure value
+     * @param[in] req_exposure new exposure
+     * @param[out] error_code error code included in response
+     * @param[out] res_msg human-readable error message included in response
+     * @return true if exposure is successfully updated
+     */
+    bool k4aUpdateExposure(int req_exposure, int& error_code, std::string& res_msg);
+    
+    /**
+     * @brief auto tune exposure with given target blue value
+     * @param[in] target_blue_value requested blue value to tune exposure to
+     * @param[out] final_exposure exposure camera is set to after call
+     * @param[out] error_code error code included in response
+     * @param[out] res_msg human-readable error message included in response
+     * @return true if auto tuning exposure is successfully completed
+     */
+    bool k4aAutoTuneExposure(int target_blue_value, int& final_exposure, int& error_code, std::string& res_msg);
+    
+    /**
+     * @brief check if dynamic_reconfigure response has correctly updated exposure
+     * @param[in] requested_exposure exposure originally requested in k4aUpdateExposure
+     * @param[out] updated_exposure exposure that was returned by the dynamic_reconfigure call
+     * @param[out] error_code error code included in response
+     * @param[out] res_msg human-readable error message included in response
+     * @return true if requested_exposure == updated_exposure
+     */
+    bool k4aCameraExposureUpdateCheck(int requested_exposure, int updated_exposure, int& error_code, std::string& res_msg);
```

Comment:
updated_exposure is an out but is passed by value? Either the docs should change or we should pass by reference (&updated_exposure)

Reply:
Oops, meant to be param[in] in documentation, good catch!